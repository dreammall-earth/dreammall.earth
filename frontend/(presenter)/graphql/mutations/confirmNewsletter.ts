import { gql } from 'graphql-tag'

export const confirmNewsletter = gql`
  mutation confirmNewsletter($code: String!) {
    confirmNewsletter(code: $code)
  }
`
