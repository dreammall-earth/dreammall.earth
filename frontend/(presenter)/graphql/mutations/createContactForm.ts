import { gql } from 'graphql-tag'

export const createContactForm = gql`
  mutation createContactForm($data: ContactFormInput!) {
    createContactForm(contactFormData: $data)
  }
`
