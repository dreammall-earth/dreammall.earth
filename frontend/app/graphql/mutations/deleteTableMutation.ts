import { gql } from 'graphql-tag'

export const deleteTableMutation = gql(`mutation DeleteTable($tableId: Int!) {
  deleteTable(tableId: $tableId)
}`)
