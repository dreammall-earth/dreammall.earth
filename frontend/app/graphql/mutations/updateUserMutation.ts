import { gql } from 'graphql-tag'

import type { CurrentUser } from '#app/stores/userStore'

export type UpdateUserInput = {
  name: string
  introduction?: string | null
  availability?: null | 'available' | 'partly_available' | 'busy'
}

export const updateUserMutation = gql`
  mutation updateUser($data: UpdateUserInput!) {
    updateUser(data: $data) {
      id
      name
      username
      introduction
      availability
      details {
        id
        category
        text
      }
      social {
        id
        type
        link
      }
      table {
        id
        name
        public
        users {
          id
          role
          name
          username
        }
      }
    }
  }
`
export type UpdateUserMutationResult = {
  updateUser: CurrentUser
}
