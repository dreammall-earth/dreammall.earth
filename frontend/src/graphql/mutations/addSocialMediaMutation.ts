import { gql } from 'graphql-tag'

export type SocialMediaType =
  | 'discord'
  | 'telegram'
  | 'snapchat'
  | 'reddit'
  | 'wechat'
  | 'whatsapp'
  | 'xing'
  | 'pintarest'
  | 'facebook'
  | 'instagram'
  | 'linkedin'
  | 'youtube'
  | 'tiktok'
  | 'x'

export type AddSocialMediaInput = {
  type: SocialMediaType
  link: string
}

export const addSocialMediaMutation = gql`
  mutation addSocialMedia($data: AddSocialMediaInput!) {
    addSocialMedia(data: $data) {
      id
      type
      link
    }
  }
`

export type AddSocialMediaMutationResult = {
  addSocialMedia: AddSocialMediaInput & {
    id: number
  }
}
