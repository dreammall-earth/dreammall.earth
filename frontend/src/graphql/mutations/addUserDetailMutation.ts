import { gql } from 'graphql-tag'

export type AddUserDetailInput = {
  text: string
  category: 'place' | 'work' | 'language' | 'education' | 'feeling'
}

export const addUserDetailMutation = gql`
  mutation addUserDetail($data: AddUserDetailInput!) {
    addUserDetail(data: $data) {
      id
      text
      category
    }
  }
`

export type AddUserDetailMutationResult = {
  addUserDetail: AddUserDetailInput & {
    id: number
  }
}
