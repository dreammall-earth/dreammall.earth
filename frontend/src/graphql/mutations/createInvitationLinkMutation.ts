import { gql } from 'graphql-tag'

export const createInvitationLinkMutation = gql`
  mutation {
    createInvitationLink
  }
`
