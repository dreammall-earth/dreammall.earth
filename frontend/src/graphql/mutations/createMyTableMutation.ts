import { gql } from 'graphql-tag'

export const createMyTableMutation = gql`
  mutation ($name: String!, $isPublic: Boolean!, $userIds: [Int]) {
    createMyRoom(name: $name, isPublic: $isPublic, userIds: $userIds) {
      id
      name
      public
      users {
        id
        role
        name
        username
      }
    }
  }
`
