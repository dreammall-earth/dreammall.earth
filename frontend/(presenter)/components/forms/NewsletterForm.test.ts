import { mount, flushPromises } from '@vue/test-utils'
import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest'

import { subscribeToNewsletter } from '#presenter/graphql/mutations/subscribeToNewsletter'
import { mockClient } from '#tests/mock.apolloClient'

import NewsletterForm from './NewsletterForm.vue'

const subscribeToNewsletterMutationMock = vi.fn()

describe('NewsletterForm', () => {
  const Wrapper = () => {
    return mount(NewsletterForm)
  }

  mockClient.setRequestHandler(
    subscribeToNewsletter,
    subscribeToNewsletterMutationMock.mockResolvedValue({ data: { subscribeToNewsletter: true } }),
  )

  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders form', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('submit', () => {
    describe('empty form', () => {
      beforeEach(async () => {
        await wrapper.find('form').trigger('submit.prevent')
        await flushPromises()
      })

      it('shows errors for all 4 fields', () => {
        const errorMessages = wrapper.findAll('.v-messages__message')
        expect(errorMessages).toHaveLength(4)
        // firstname
        expect(errorMessages[0].text()).toBe('Dieses Feld wird benötigt.')
        // lastname
        expect(errorMessages[1].text()).toBe('Dieses Feld wird benötigt.')
        // email
        expect(errorMessages[2].text()).toBe('Dieses Feld wird benötigt.')
        // checkbox
        expect(errorMessages[3].text()).toBe("$t('validation.fieldRequired')")
      })

      it('user feedback not visible', () => {
        expect(wrapper.find('span.info-text.form-success').exists()).toBe(false)
        expect(wrapper.find('span.info-text.form-error').exists()).toBe(false)
      })

      it('does not call the API', () => {
        expect(subscribeToNewsletterMutationMock).not.toHaveBeenCalled()
      })
    })

    describe('valid form', () => {
      beforeEach(async () => {
        vi.useFakeTimers()
        await wrapper.find('input[name="firstname"]').setValue('Peter')
        await wrapper.find('input[name="lastname"]').setValue('Lustig')
        await wrapper.find('input[name="email"]').setValue('peter@lustig.de')
        await wrapper.find('input[name="dataprivacy"]').setValue(true)
      })

      afterEach(() => {
        vi.useRealTimers()
      })

      describe('with success', () => {
        beforeEach(async () => {
          await wrapper.find('form').trigger('submit.prevent')
          await flushPromises()
        })

        it('calls the API', () => {
          expect(subscribeToNewsletterMutationMock).toHaveBeenCalledWith({
            data: {
              firstName: 'Peter',
              lastName: 'Lustig',
              email: 'peter@lustig.de',
            },
          })
        })

        it('resets the form', () => {
          expect(wrapper.find('input[name="firstname"]').element).toHaveProperty('value', '')
          expect(wrapper.find('input[name="lastname"]').element).toHaveProperty('value', '')
          expect(wrapper.find('input[name="email"]').element).toHaveProperty('value', '')
          expect(wrapper.find('input[name="dataprivacy"]').element).toHaveProperty('checked', false)
        })

        describe('success message for user', () => {
          it('shows message', () => {
            expect(wrapper.find('span.info-text.form-success').exists()).toBe(true)

            expect(wrapper.find('span.info-text.form-success').text()).toBe(
              "$t('home.newsletterSection.newsletterForm.successMsg')",
            )
          })

          describe('run timers', () => {
            beforeEach(() => {
              vi.runAllTimers()
            })

            it('does not show the message anymore', () => {
              expect(wrapper.find('span.info-text.form-success').exists()).toBe(false)
            })
          })
        })
      })

      describe('with error', () => {
        beforeEach(async () => {
          subscribeToNewsletterMutationMock.mockRejectedValue({ message: 'Ouch!' })
          await wrapper.find('form').trigger('submit.prevent')
          await flushPromises()
        })

        it('calls the API', () => {
          expect(subscribeToNewsletterMutationMock).toHaveBeenCalledWith({
            data: {
              firstName: 'Peter',
              lastName: 'Lustig',
              email: 'peter@lustig.de',
            },
          })
        })

        it('does not reset the form', () => {
          expect(wrapper.find('input[name="firstname"]').element).toHaveProperty('value', 'Peter')
          expect(wrapper.find('input[name="email"]').element).toHaveProperty(
            'value',
            'peter@lustig.de',
          )
          expect(wrapper.find('input[name="lastname"]').element).toHaveProperty('value', 'Lustig')

          expect(wrapper.find('input[name="dataprivacy"]').element).toHaveProperty('checked', true)
        })

        describe('error message for user', () => {
          it('shows error message', () => {
            expect(wrapper.find('span.info-text.form-error').exists()).toBe(true)
            expect(wrapper.find('span.info-text.form-error').text()).toBe(
              "$t('home.newsletterSection.newsletterForm.errorMsg')",
            )
          })

          describe('run timers', () => {
            beforeEach(() => {
              vi.runAllTimers()
            })

            it('does not show the message anymore', () => {
              expect(wrapper.find('span.info-text.form-error').exists()).toBe(false)
            })
          })
        })
      })
    })
  })
})
