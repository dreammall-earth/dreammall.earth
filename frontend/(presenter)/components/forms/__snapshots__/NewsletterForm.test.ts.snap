// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewsletterForm > renders form 1`] = `
<div
  class="v-sheet v-theme--light bg-transparent"
  data-v-72f5b408=""
>
  <form
    class="v-form newsletter-form"
    data-v-72f5b408=""
    novalidate=""
  >
    
    <div
      class="v-row"
      data-v-72f5b408=""
    >
      <div
        class="v-col-sm-6 v-col-12"
        data-v-72f5b408=""
      >
        <div
          class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field newsletter-text"
          data-v-72f5b408=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
              style="background-color: rgba(174, 179, 189, 0.50);"
            >
              <div
                class="v-field__overlay"
              />
              <div
                class="v-field__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="background-color: #3D4753; color: #fff; caret-color: #fff; opacity: NaN;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="background-color: #3D4753; color: #fff; caret-color: #fff; opacity: NaN; width: 0%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                        style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                        style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div
                class="v-field__field"
                data-no-activator=""
              >
                <label
                  aria-hidden="true"
                  class="v-label v-field-label v-field-label--floating"
                  for="input-0"
                >
                  <!---->
                  
                  $t('home.newsletterSection.newsletterForm.firstname')
                  
                </label>
                <label
                  class="v-label v-field-label"
                  for="input-0"
                >
                  <!---->
                  
                  $t('home.newsletterSection.newsletterForm.firstname')
                  
                </label>
                
                
                <!---->
                <input
                  aria-describedby="input-0-messages"
                  class="v-field__input"
                  id="input-0"
                  name="firstname"
                  required=""
                  size="1"
                  type="text"
                  value=""
                />
                <!---->
                
                
              </div>
              <!---->
              <!---->
              <div
                class="v-field__outline"
              >
                <!---->
                <!---->
              </div>
            </div>
            
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <div
        class="v-col-sm-6 v-col-12"
        data-v-72f5b408=""
      >
        <div
          class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field newsletter-text"
          data-v-72f5b408=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
              style="background-color: rgba(174, 179, 189, 0.50);"
            >
              <div
                class="v-field__overlay"
              />
              <div
                class="v-field__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="background-color: #3D4753; color: #fff; caret-color: #fff; opacity: NaN;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="background-color: #3D4753; color: #fff; caret-color: #fff; opacity: NaN; width: 0%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                        style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                        style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div
                class="v-field__field"
                data-no-activator=""
              >
                <label
                  aria-hidden="true"
                  class="v-label v-field-label v-field-label--floating"
                  for="input-2"
                >
                  <!---->
                  
                  $t('home.newsletterSection.newsletterForm.lastname')
                  
                </label>
                <label
                  class="v-label v-field-label"
                  for="input-2"
                >
                  <!---->
                  
                  $t('home.newsletterSection.newsletterForm.lastname')
                  
                </label>
                
                
                <!---->
                <input
                  aria-describedby="input-2-messages"
                  class="v-field__input"
                  id="input-2"
                  name="lastname"
                  required=""
                  size="1"
                  type="text"
                  value=""
                />
                <!---->
                
                
              </div>
              <!---->
              <!---->
              <div
                class="v-field__outline"
              >
                <!---->
                <!---->
              </div>
            </div>
            
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <div
      class="v-row"
      data-v-72f5b408=""
    >
      <div
        class="v-col-sm-6 v-col-12"
        data-v-72f5b408=""
      >
        <div
          class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field newsletter-text"
          data-v-72f5b408=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
              style="background-color: rgba(174, 179, 189, 0.50);"
            >
              <div
                class="v-field__overlay"
              />
              <div
                class="v-field__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="background-color: #3D4753; color: #fff; caret-color: #fff; opacity: NaN;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="background-color: #3D4753; color: #fff; caret-color: #fff; opacity: NaN; width: 0%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                        style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                        style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div
                class="v-field__field"
                data-no-activator=""
              >
                <label
                  aria-hidden="true"
                  class="v-label v-field-label v-field-label--floating"
                  for="input-4"
                >
                  <!---->
                  
                  $t('home.newsletterSection.newsletterForm.email')
                  
                </label>
                <label
                  class="v-label v-field-label"
                  for="input-4"
                >
                  <!---->
                  
                  $t('home.newsletterSection.newsletterForm.email')
                  
                </label>
                
                
                <!---->
                <input
                  aria-describedby="input-4-messages"
                  class="v-field__input"
                  id="input-4"
                  name="email"
                  required=""
                  size="1"
                  type="email"
                  value=""
                />
                <!---->
                
                
              </div>
              <!---->
              <!---->
              <div
                class="v-field__outline"
              >
                <!---->
                <!---->
              </div>
            </div>
            
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <div
        class="v-col-sm-6 d-flex align-center v-col-12"
        data-v-72f5b408=""
      >
        <button
          class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--form-submit main-button-auto w-100"
          data-v-72f5b408=""
          data-v-f4a272bd=""
          style="background-color: #23ad5b; color: #fff; caret-color: #fff; width: auto; height: auto;"
          type="submit"
        >
          
          <span
            class="v-btn__overlay"
          />
          <span
            class="v-btn__underlay"
          />
          
          <!---->
          <span
            class="v-btn__content"
            data-no-activator=""
          >
            
            
            <!--v-if-->
            <span
              class="main-button-content"
              data-v-f4a272bd=""
            >
              $t('home.newsletterSection.newsletterForm.submitBtn') 
              <!--v-if-->
            </span>
            
            
          </span>
          <!---->
          <!---->
        </button>
      </div>
    </div>
    <div
      class="v-row"
      data-v-72f5b408=""
    >
      <div
        class="v-col v-col-12 d-flex align-start"
        data-v-72f5b408=""
      >
        <div
          class="v-input v-input--horizontal v-input--density-compact v-theme--light v-locale--is-ltr v-input--dirty v-checkbox"
          data-v-72f5b408=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-selection-control v-selection-control--density-compact v-checkbox-btn"
            >
              <div
                class="v-selection-control__wrapper"
              >
                <!---->
                <div
                  class="v-selection-control__input"
                >
                  
                  <i
                    aria-hidden="true"
                    class="mdi-checkbox-blank-outline mdi v-icon notranslate v-theme--light v-icon--size-default"
                  />
                  <input
                    aria-describedby="checkbox-7-messages"
                    aria-disabled="false"
                    id="checkbox-7"
                    name="dataprivacy"
                    required=""
                    type="checkbox"
                    value="true"
                  />
                  
                </div>
              </div>
              <!---->
            </div>
            
          </div>
          <!---->
          <!---->
        </div>
        <span
          class="ml-6 ml-sm-4 pt-2 newsletter-dataprivacy"
          data-v-72f5b408=""
        >
          $t('home.newsletterSection.newsletterForm.privacy') 
          <a
            class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain anchor-link anchor-link--text dataprivacy-link"
            data-v-72f5b408=""
            data-v-bbcf5fdc=""
            href="/datenschutz"
          >
            
            <span
              class="v-btn__overlay"
            />
            <span
              class="v-btn__underlay"
            />
            
            <!---->
            <span
              class="v-btn__content"
              data-no-activator=""
            >
              
              
              $t('home.newsletterSection.newsletterForm.privacyLinkLabel')
              
              
            </span>
            <!---->
            <!---->
          </a>
        </span>
      </div>
    </div>
    <div
      class="v-row align-content-center mt-0"
      data-v-72f5b408=""
    >
      <div
        class="v-col pt-0"
        data-v-72f5b408=""
      >
        <transition-group-stub
          appear="false"
          css="true"
          data-v-72f5b408=""
          persisted="false"
        >
          <!--v-if-->
          <!--v-if-->
        </transition-group-stub>
      </div>
    </div>
    
  </form>
</div>
`;
